src/components/organisms/NewProfileModal/components/ProfileModalFetchUser/useCurrentModalUser.tsx

```diff
@@ -0,0 +1,27 @@
+import { User } from "types/User";
+
+import { WithId } from "utils/id";
+import { currentModalUserSelector } from "utils/selectors";
+
+import { isLoaded, useFirestoreConnect } from "hooks/useFirestoreConnect";
+import { useSelector } from "hooks/useSelector";
+
+const useConnectCurrentModalUser = (userId?: string) => {
+  useFirestoreConnect([
+    {
+      collection: "users",
+      doc: userId ?? "",
+      storeAs: "currentModalUser",
+    },
+  ]);
+};
+
+export const useCurrentModalUser = (
+  userId?: string
+): [WithId<User> | undefined, boolean] => {
+  useConnectCurrentModalUser(userId);
+
+  const user = useSelector(currentModalUserSelector);
+
+  return [user?.[0], isLoaded(user)];
```

Comment:
If you use currentModalUserDataSelector, you won't have to to `user?.[0]`

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/2291/commits/a1bc6fb1274bf1267476d08be3f705cfdb35ca63